### PR TITLE
Refactored MessageLogBehavior

### PR DESCRIPTION
**Pull request by JakenVeina**
Refactored MessageLogBehavior into MessageLoggingBehavior and MessageTrackingBehavior.

Separation of logic helps reduce the complexity of the methods involved, and highlights the fact that these behaviors are really separate units of business functionality within MODiX that have no need to be related to each other.

Refactored behaviors are implemented through `INotificationHandler` rather than deprecated `BehaviorBase`.

Refactored behaviors use proper types from `Discord.Net.Abstractions`.

Refactored behaviors fully support trace logging.

Refactored behaviors are fully unit tested.

### CODE REVIEW

File: Modix.Services/MessageLogging/MessageLoggingLogMessages.cs [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        public static void MessageDeletedHandling(
+                ILogger logger)
+            => _messageDeletedHandling.Invoke(
+                logger);
+        private static readonly Action<ILogger> _messageDeletedHandling
+            = LoggerMessage.Define(
+                    LogLevel.Information,
+                    EventType.MessageDeletedHandling.ToEventId(),
+                    "Deleted message handled")
+                .WithoutException();
```
Comment (Scott-Caldwell) [inline Modix.Services/MessageLogging/MessageLoggingLogMessages.cs:109]: Is the text for these messages swapped? `MessageDeletedHandled` says "handling" and `MessageDeletedHandling` says "handled"

Comment (JakenVeina) [reply Modix.Services/MessageLogging/MessageLoggingLogMessages.cs:109]: Yup.

Comment (Scott-Caldwell) [reply Modix.Services/MessageLogging/MessageLoggingLogMessages.cs:109]: (also applies to the update ones below)

Comment (JakenVeina) [reply Modix.Services/MessageLogging/MessageLoggingLogMessages.cs]: Goddamnit

**Pull request merged**